pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -107,3 +108,29 @@ def attach_model_logging_functions(self, model):
         for callback in self.trainer.callbacks:
             callback.log = model.log
             callback.log_dict = model.log_dict
+
+    def attach_model_callbacks(self, model: LightningModule) -> None:
+        """
+        Attaches the callbacks defined in the model.
+        If a callback returned by the model's configure_callback method has the same type as one or several
+        callbacks already present in the trainer callbacks list, it will replace them.
+
+        Args:
+            A model which may or may not define new callbacks in
+            :meth:`~pytorch_lightning.core.lightning.LightningModule.configure_callbacks`.
+        """
+        model_callbacks = model.configure_callbacks()
+        model_callback_types = set(type(c) for c in model_callbacks)
```

Comment:
you can short-circuit if model_callbacks empty

Reply:
luckily I had some wire here at home. Done!